### ISSUE TITLE
ARM64 Builds

### ISSUE THREAD

**Issue by TCB13** [veracrypt/VeraCrypt#846]
Hello,

I noticed that https://www.veracrypt.fr/en/Downloads.html doesn't include ARM64 build. The included armhf build might run on the popular raspberry pi however there are now many SBCs that are purely ARM64 and can't install this package properly.

For instance on a NanoPi M4v2 (RK3399 ARM64) running Armbian you'll get a conflict between the sudo of the system and the armhf version that the package requires:

``````
$ dpkg -i veracrypt-console-1.24-Update7-Debian-10-armhf.deb
Selecting previously unselected package veracrypt-console:armhf.
(Reading database ... 54806 files and directories currently installed.)
Preparing to unpack veracrypt-console-1.24-Update7-Debian-10-armhf.deb ...
Unpacking veracrypt-console:armhf (1.24.23-1) ...
dpkg: dependency problems prevent configuration of veracrypt-console:armhf:
 veracrypt-console:armhf depends on libfuse2.
 veracrypt-console:armhf depends on sudo.

dpkg: error processing package veracrypt-console:armhf (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 veracrypt-console:armhf

$ apt install sudo libfuse2
Reading package lists... Done
Building dependency tree
Reading state information... Done
libfuse2 is already the newest version (2.9.9-1+deb10u1).
sudo is already the newest version (1.8.27-1+deb10u3).
You might want to run 'apt --fix-broken install' to correct these.
The following packages have unmet dependencies:
 veracrypt-console:armhf : Depends: libfuse2:armhf but it is not going to be installed
                           Depends: sudo:armhf
E: Unmet dependencies. Try 'apt --fix-broken install' with no packages (or specify a solution).
``````

Are there any plans for pure ARM64 builds? Or details on how to properly build it from the source?

Thank you.

**Comment by unit193** [comment]
Hello,

If you're fine with third party sources, there are arm builds that I've uploaded (uploaded the source, OBS does the build) to [this](https://build.opensuse.org/package/show/home:unit193:veracrypt/veracrypt) OBS repo.  You can verify the source tarball against the fingerprint provided to confirm the source is the same.  If not, then no hard feelings and I hope the build is otherwise provided soon.

~Unit 193

**Comment by TCB13** [comment]
Hello,

Thank you for the link. I'm mostly fine with third party sources and I'm going to give a try. However it seems odd with the amount of ARM64 devices that are around now that there aren't official builds on the website yet.

Have a nice day.

**Comment by dorandeluz** [comment]
Tried building VeraCrypt_1.24-Update7 from source for arm64. Everything goes well until almost the end:

```
$ make NOGUI=1 WXSTATIC=1
...
Compiling Main.cpp
Converting License.txt
Converting Language.xml
Compiling Resources.cpp
Linking veracrypt
free(): invalid pointer
Aborted (core dumped)
make[1]: *** [Main.make:168: veracrypt] Error 1
make: *** [Makefile:424: all] Error 2
```
Executable is created but fails too:
```
$ ./Main/veracrypt
free(): invalid pointer
Aborted (core dumped)
```

Ubuntu 20.04.3 LTS (GNU/Linux 5.4.0-1046-raspi aarch64)
Raspberry Pi 4B (8GB, ARMv8)

Any idea on what I could try?

**Comment by idrassi** [comment]
I was able to get access to a Raspberry Pi 4 so I will be building official ARM64 packages of VeraCrypt.
@dorandeluz I will perform a build tomorrow under Debian 11 on the Raspberry Pi 4 and I will let you know if I reproduce the crash. Maybe the issue is linked to wxWodgets because it is not clear if you used version 3.0.5 as in the official build scripts.

**Comment by idrassi** [comment]
I was able to successfully build VeraCrypt using latest sources from master for ARM64 on Raspberry PI aarch64 running Debian 11. So there must be some other issues with your setup.

I'm attaching a zip containing ARM .deb packages for VeraCrypt that I created on Debian 11 and which will also work on Ubuntu 20.04. The .deb files are signed by official VeraCrypt PGP key and the signatures are included in the zip.

[veracrypt-1.25.4-arm64-Debian11.zip](https://github.com/veracrypt/VeraCrypt/files/7638111/veracrypt-1.25.4-arm64-Debian11.zip)
